src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1336,6 +1455,73 @@ public IEnumerable<(string Key, long Index)> GetCompareExchangeIndexes(Transacti
             }
         }
 
+        public IEnumerable<(string Key, long Index, BlittableJsonReaderObject Value)> GetCompareExchangeFrom(TransactionOperationContext context, string dbName, long fromIndex, int take)
+        {
+            CompareExchangeCommandBase.GetPrefixIndexSlices(context.Allocator, dbName, fromIndex, out var indexTuple);
+
+            using (indexTuple.Scope)
+            {
+                var table = context.Transaction.InnerTransaction.OpenTable(CompareExchangeSchema, CompareExchange);
+                using (Slice.External(context.Allocator, indexTuple.Buffer, indexTuple.Buffer.Length, out var keySlice))
+                    foreach (var tvr in table.SeekForwardFrom(CompareExchangeSchema.Indexes[CompareExchangeIndexSlice], keySlice, 0))
+                    {
+                        if (take-- <= 0)
+                            yield break;
+
+                        var key = ReadCompareExchangeKey(tvr.Result.Reader, dbName);
+                        var index = ReadCompareExchangeIndex(tvr.Result.Reader);
+                        var value = ReadCompareExchangeValue(context, tvr.Result.Reader);
+
+                        yield return (key, index, value);
+                    }
+            }
+        }
+
+        public IEnumerable<(string key, long index, BlittableJsonReaderObject value)> GetCompareExchangeFromPrefix(TransactionOperationContext context, string dbName, long fromIndex, int take)
+        {
+            CompareExchangeCommandBase.GetPrefixIndexSlices(context.Allocator, dbName, fromIndex, out var indexTuple);
+
+            using (indexTuple.Scope)
+            {
+                var table = context.Transaction.InnerTransaction.OpenTable(CompareExchangeSchema, CompareExchange);
+                using (Slice.External(context.Allocator, indexTuple.Buffer, indexTuple.Buffer.Length, out var keySlice))
+                using (Slice.External(context.Allocator, indexTuple.Buffer, indexTuple.Buffer.Length - sizeof(long), out var prefix))
+                    foreach (var tvr in table.SeekForwardFromPrefix(CompareExchangeSchema.Indexes[CompareExchangeIndexSlice], keySlice, prefix, 0))
```

Comment:
Here you are properly ensuring that we can get data only within the same db, should be done elsewhere as well

Reply:
done